catchup/universalFetcher.go

```diff
@@ -90,21 +90,21 @@ func (uf *universalBlockFetcher) fetchBlock(ctx context.Context, round basics.Ro
 	return block, cert, downloadDuration, err
 }
 
-func processBlockBytes(fetchedBuf []byte, r basics.Round, debugStr string) (blk *bookkeeping.Block, cert *agreement.Certificate, err error) {
+func processBlockBytes(fetchedBuf []byte, r basics.Round, peerAddr string) (blk *bookkeeping.Block, cert *agreement.Certificate, err error) {
 	var decodedEntry rpcs.EncodedBlockCert
 	err = protocol.Decode(fetchedBuf, &decodedEntry)
 	if err != nil {
-		err = fmt.Errorf("fetchBlock(%d): cannot decode block from peer %v: %v", r, debugStr, err)
+		err = makeErrCannotDecodeBlock(r, peerAddr, err)
```

Comment:
it is interesting, since `error` is interface, I thought one needs to return a pointer to a struct. But attaching `Error()`  to the type, not pointer looks working, nice.

Reply:
looks like a pointer or struct can be passed for an interface both for return type, or a function parameter.